internal/generator/vector/conf/conf.go

```diff
@@ -70,6 +71,7 @@ func Conf(secrets map[string]*corev1.Secret, clfspec obs.ClusterLogForwarderSpec
 
 	outputMap := map[string]*output.Output{}
 	for _, spec := range clfspec.Outputs {
+		op[framework.OptionForwarderName] = forwarderName
```

Comment:
Options are maps which are pass by ref;  I believe this is setting the same value over and over on the same map.  You should be able to move this out of the loop